distributed/worker.py

```diff
@@ -2149,8 +2148,7 @@ def transition_cancelled_error(
                 traceback_text,
                 stimulus_id=stimulus_id,
             )
-        if ts._next:
-            recs[ts] = ts._next
+        recs[ts] = "released"
```

Comment:
Why override what the transition function we just called recommended to do?

Reply:
Good catch. This entire transition is a bit off for "historic" reasons (i.e. I got confused during the initial refactor). The cancellation->error transition can only be reached from an executing state, i.e. previous is always executing and the "flight" path is dead code.
Further, this error transition will *never* generate a recommendation for this key since error is supposed to be dead end unless the scheduler releases the key. However, in this specific situation, the scheduler _already_ released the key which is why we'll need to recommend the release of this key.

I'll updated this transition accordingly and added two tests for it